### PR TITLE
Use HF datasets

### PR DESCRIPTION
**Pull request by salma-remyx**
Instead of processing a local directory of images, we want to process from huggingface datasets which contain `PIL.Image` images that we can process. This makes it easier to use open source datasets that don't have to be wrangled locally.

This new pipeline design will also push new dataset to hf hub

Test by running:
`bash run.sh`

Test results from config.yaml setup: https://huggingface.co/datasets/salma-remyx/vqasynth_sample_processed?row=0

### CODE REVIEW

File: config/config.yaml [hunk 1]
```diff
@@ -1,9 +1,11 @@
 # config.yaml
 directories:
-  image_dir: /home/ubuntu/vqasynth_sample
   output_dir: /home/ubuntu/vqasynth_output
 
 arguments:
+  source_repo_id: remyxai/vqasynth_sample
+  target_repo_id: salma-remyx/vqasynth_sample_processed
```
Comment (smellslikeml) [inline config/config.yaml:7]: source_id, result_id?
should you code in the HF username?

File: config/config.yaml [hunk 2]
```diff
@@ -1,9 +1,11 @@
...
+  image_col: "image"
```
Comment (smellslikeml) [inline config/config.yaml:8]: something more explicit or descriptive than image_col

File: docker/filter_stage/process_filter.py [hunk 3]
```diff
@@ -3,42 +3,42 @@
...
-            )
+    dataset = dataset.map(process_row)
 
-            df['keep'] = df.apply(
-                lambda row: tag_filter.filter_by_tag(
-                    row['tag'], include_tags, exclude_tags
-                ),
-                axis=1
-            )
+    dataset_filtered = dataset.filter(lambda example: example['keep'])
```
Comment (smellslikeml) [inline docker/filter_stage/process_filter.py:30]: is keep a good field name

**Review by smellslikeml — APPROVED** [review]

File: config/config.yaml (hunk 2 above)
Comment (salma-remyx) [reply config/config.yaml:8]: @smellslikeml do you want `image_data_field` or `image_data_column`

**Pull request merged**